router/failed-events-manager.go

```diff
@@ -0,0 +1,182 @@
+		for _, failedEvent := range failedEvents {
+			_, err = stmt.Exec(failedEvent.DestinationID, failedEvent.RecordID, createdAt)
+			if err != nil {
+				panic(err)
+			}
+		}
+		_, err = stmt.Exec()
+		if err != nil {
+			panic(err)
+		}
+		stmt.Close()
+	}
+}
+
+func (fem *FailedEventsManagerT) DropFailedRecordIDs(taskRunID string) {
+	if !failedKeysEnabled {
+		return
+	}
+	//Drop table
+	table := fmt.Sprintf(`%s_%s`, failedKeysTablePrefix, taskRunID)
+	sqlStatement := fmt.Sprintf(`DROP TABLE IF EXISTS %s`, table)
+	_, err := fem.dbHandle.Exec(sqlStatement)
+	if err != nil {
+		pkgLogger.Errorf("Failed to drop table %s with error: %v", taskRunID, err)
+	}
+}
+
+func (fem *FailedEventsManagerT) FetchFailedRecordIDs(taskRunID string) []*FailedEventRowT {
+	if !failedKeysEnabled {
+		return []*FailedEventRowT{}
+	}
+	failedEvents := make([]*FailedEventRowT, 0)
+
+	var rows *sql.Rows
+	var err error
+	table := fmt.Sprintf(`%s_%s`, failedKeysTablePrefix, taskRunID)
+	sqlStatement := fmt.Sprintf(`SELECT %[1]s.destination_id, %[1]s.record_id
+                                             FROM %[1]s `, table)
+	rows, err = fem.dbHandle.Query(sqlStatement)
+	if err != nil {
+		pkgLogger.Errorf("Failed to fetch from table %s with error: %v", taskRunID, err)
+		return failedEvents
+	}
+	defer rows.Close()
+
+	for rows.Next() {
+		var failedEvent FailedEventRowT
+		err := rows.Scan(&failedEvent.DestinationID, &failedEvent.RecordID)
+		if err != nil {
+			panic(err)
+		}
+		failedEvents = append(failedEvents, &failedEvent)
+	}
+
+	return failedEvents
+}
+
+func CleanFailedRecordsTableProcess() {
+	if !failedKeysEnabled { // TODO: is this required or let the cron run?
+		return
+	}
+	for {
+		dbHandle, err := sql.Open("postgres", jobsdb.GetConnectionString())
+		if err != nil {
+			panic(err)
+		}
+		failedKeysLike := failedKeysTablePrefix + "%"
+		failedKeysTableQuery := fmt.Sprintf(`SELECT table_name
+													FROM information_schema.tables
+													WHERE table_schema='public' AND table_type='BASE TABLE AND table_name ilike '%s'`, failedKeysLike)
+		rows, err := dbHandle.Query(failedKeysTableQuery)
+		if err != nil {
+			panic(err)
+		}
+		for rows.Next() {
+			var table string
+			err = rows.Scan(&table)
+			if err != nil && err != sql.ErrNoRows {
+				panic(err)
+			}
+			latestCreatedAtQuery := fmt.Sprintf(`SELECT created_at from %s order by created_at desc limit 1`, table)
+			row := dbHandle.QueryRow(latestCreatedAtQuery)
+			var latestCreatedAt time.Time
+			err = row.Scan(&latestCreatedAtQuery)
+			if err != nil && err != sql.ErrNoRows {
+				panic(err)
+			}
+			currentTime := time.Now()
+			diff := currentTime.Sub(latestCreatedAt)
+			if diff > failedKeysExpire {
+				dropQuery := fmt.Sprintf(`DROP TABLE IF EXISTS %s`, table)
+				rows, err = dbHandle.Query(dropQuery)
+				if err != nil {
+					pkgLogger.Errorf("Failed to drop table %s with error: %v", table, err)
+				}
+			}
+		}
+		dbHandle.Close()
+		time.Sleep(24 * time.Hour)
```

Comment:
Is this 24hrs or should we be making this configurable?

Reply:
made it configurable